node/node.go

```diff
@@ -148,6 +148,48 @@ type Node struct {
 	prometheusSrv    *http.Server
 }
 
+func createExternalPrivValidator(listenAddr string, logger log.Logger) (types.PrivValidator, error) {
+	protocol, address := cmn.ProtocolAndAddress(listenAddr)
+
+	var pvsc types.PrivValidator
+
+	switch (protocol) {
+	case "unix":
+		pvsc = privval.NewIPCVal(
+			logger.With("module", "privval"),
+			address,
+		)
+
+	case "tcp":
+		// TODO: persist this key so external signer
+		// can actually authenticate us
+		pvsc = privval.NewTCPVal(
+			logger.With("module", "privval"),
+			listenAddr,
+			ed25519.GenPrivKey(),
+		)
+
+	default:
+		return nil, fmt.Errorf(
+			"Error creating private validator: expected either tcp or unix "+
+			"protocols, got %s",
+			protocol,
+		)
+	}
+
+	pvServ, ok := pvsc.(cmn.Service)
+	if !ok {
```

Comment:
let's remove this check. It's clearly a developer's error and it's OK to panic if priv validator client does not implement Service.

Reply:
Sure thing.